### PR TITLE
Bug fixes.

### PR DESCRIPTION
**Pull request by sudswasavi**
Fixes a few bugs like:

- Get the kubernetes updates first and then get the Istio updates
so that the Secrets for ISTIO_MUTUAL can be created before any of the
istio resources are processed.

- Misc fixes

### CODE REVIEW

File: amc/pkg/istio/nodes/avi_model_translator.go [hunk 1]
```diff
@@ -635,7 +641,7 @@ func (o *AviObjectGraph) evaluateTLSPools(ns string, randString string, destinat
 			// There's no endpoint object for the service.
 			poolNode.Servers = nil
 		} else {
-			poolNode.Servers = o.extractServers(epObj, portNumber, portName, destination.Destination.Subset, ns, labels)
+			o.extractServers(epObj, portNumber, portName, destination.Destination.Subset, ns, labels, poolNode)
```
Comment (monotosh-avi) [inline amc/pkg/istio/nodes/avi_model_translator.go]: We are setting poolNode.Port both in extractServers() and bellow, is that fine ?

File: amc/pkg/istio/rest/dequeue_nodes.go [hunk 2]
```diff
@@ -37,9 +37,9 @@ func DeQueueNodes(key string) {
 		if keysplit[1] == "ISTIO_MUTUAL_SECRET" {
 			if avimodelIntf != nil {
 				sslNodesModel := avimodelIntf.(*nodes.AviObjectGraph)
-				processSSLKeyPkiProfile(keysplit[0], keysplit[1], sslNodesModel, cache)
+				processSSLKeyPkiProfile(keysplit[0], keysplit[2], sslNodesModel, cache)
```
Comment (adityats) [inline amc/pkg/istio/rest/dequeue_nodes.go]: pl add a comment of the keysplit contents at line 35

File: amc/pkg/istio/nodes/avi_model_translator.go (hunk 1 above)
Comment (sudswasavi) [reply amc/pkg/istio/nodes/avi_model_translator.go]: Done.

File: amc/pkg/istio/rest/dequeue_nodes.go (hunk 2 above)
Comment (sudswasavi) [reply amc/pkg/istio/rest/dequeue_nodes.go]: done

**Pull request merged**